// validator.ts/types/SanitizeTypes.js
"use strict";
/**
 * Sanitization types.
 */
(function (SanitizeTypes) {
    SanitizeTypes[SanitizeTypes["BLACKLIST"] = 1] = "BLACKLIST";
    SanitizeTypes[SanitizeTypes["ESCAPE"] = 2] = "ESCAPE";
    SanitizeTypes[SanitizeTypes["LTRIM"] = 3] = "LTRIM";
    SanitizeTypes[SanitizeTypes["NORMALIZE_EMAIL"] = 4] = "NORMALIZE_EMAIL";
    SanitizeTypes[SanitizeTypes["RTRIM"] = 5] = "RTRIM";
    SanitizeTypes[SanitizeTypes["STRIP_LOW"] = 6] = "STRIP_LOW";
    SanitizeTypes[SanitizeTypes["TO_BOOLEAN"] = 7] = "TO_BOOLEAN";
    SanitizeTypes[SanitizeTypes["TO_DATE"] = 8] = "TO_DATE";
    SanitizeTypes[SanitizeTypes["TO_FLOAT"] = 9] = "TO_FLOAT";
    SanitizeTypes[SanitizeTypes["TO_INT"] = 10] = "TO_INT";
    SanitizeTypes[SanitizeTypes["TO_STRING"] = 11] = "TO_STRING";
    SanitizeTypes[SanitizeTypes["TRIM"] = 12] = "TRIM";
    SanitizeTypes[SanitizeTypes["WHITELIST"] = 13] = "WHITELIST";
    SanitizeTypes[SanitizeTypes["CUSTOM_SANITIZATION"] = 9999] = "CUSTOM_SANITIZATION";
})(exports.SanitizeTypes || (exports.SanitizeTypes = {}));
var SanitizeTypes = exports.SanitizeTypes;
